packages/cli/src/util/colorizer.ts

```diff
@@ -0,0 +1,23 @@
+import { mapValues, isArray } from 'lodash';
+import { Dictionary } from '@stoplight/types';
+
+export type ValuesTransformer = (values: Dictionary<unknown>) => Dictionary<string | string[]>;
+
+export const PRE_PARAM_VALUE_TAG = '~pre~';
+export const POST_PARAM_VALUE_TAG = '~post~';
+
+export const transformPathParamsValues = ((
+  taggedParamsValues: RegExp,
+  path: string,
+  transform: (aString: string) => string
+): string => {
+  return path.replace(taggedParamsValues, transform('$2'));
+}).bind({}, new RegExp(`(${PRE_PARAM_VALUE_TAG},)(.*?)(,${POST_PARAM_VALUE_TAG})`, 'gm'));
+
+export const attachTagsToParamsValues: ValuesTransformer = values => {
+  return mapValues(values, attachPrePostTags);
+};
+
+const attachPrePostTags = (paramValue: unknown) => {
+  return [PRE_PARAM_VALUE_TAG].concat(isArray(paramValue) ? paramValue : `${paramValue}`).concat(POST_PARAM_VALUE_TAG);
```

Comment:
That's tough. I think a solution could be this:
1. forking `uri-template-lite`
2. changing the signature of `.expand` from `function expand(template, data)` to `function expand(template, data, cb)` and invoke `cb` on a generated value for each of the passed params
3. so it would be sth like `.expand(template, data, chalk.bold.red)`
this way we have an encoded path with the expanded parameters, which values are colored (since the coloring info itself was not escaped) 
WDYT @StefanDywersant, @XVincentX ?

Reply:
If there is any "nice to have" functionality being considered here, please keep in mind that getting PRs to other forks merged and released can take a long time, so go with a lesser solution until that's done. In general it is better to have "ok" now and "awesome" later, than to try and maintain yet another fork.

Reply:
@lag-of-death Yeah I was thinking about the same solution as well but I think we should not go in such direction

1. This is a job for the URI Template library in general
2. This is something that should be happening in Prism

In theory, instead of forking the library, we could try to look for another one that instead of simply matching and expanding would give us access to the matching data that we could use to reconstruct the URL somehow.

I think already in such state this improvement will provide a lot of value to the users, so I suggest to create a low-priority followup issue that we can work on with the passing of the time.

Reply:
@lag-of-death now that I think it could be solved by wrapping in ~pre~/~post~ each value of array.

More or less sth like that:
```suggestion
  return isArray(paramValue) ? paramValue.map(v => `${PRE_PARAM_VALUE_TAG}${v}${POST_PARAM_VALUE_TAG}`) : `${PRE_PARAM_VALUE_TAG}${paramValue}${POST_PARAM_VALUE_TAG}`;
```

This way we should colorize only the actual values, not the stuff around.

Another option worth checking is to wrap whole uri templates into colorization tags. See here: https://github.com/stoplightio/prism/blob/feature/colorize-params-values/packages/cli/src/util/paths.ts#L180. If you do `~pre~{?${formExplodedParams}*}~post~` then it should also work.

The bright side is: it is possible that expand will not touch special characters, so you could possibly put colorizing tags immediately (worth checking!).
The dark side is: it won't colorize values only, for some complex examples (matrix & explode) it will colorize bit more (**;id=1;id=2;id=3** instead of ;id=**1**;id=**2**;id=**3**).

I'd check above.

----

@XVincentX @philsturgeon  I think it is worth solving because leaving it as is will introduce bug with the wrong path being generated. However, those two solutions described above should be fairly easy to implement without forking external libs.

Reply:
@StefanDywersant, I went for the first proposed solution, thanks! :+1: